src/nbl/video/COpenGLImageView.h

```diff
@@ -53,15 +56,20 @@ class COpenGLImageView final : public IGPUImageView
 					continue;
 				swizzle[i] = ComponentMappingToGLenumSwizzle[currentMapping];
 			}
-			COpenGLExtensionHandler::extGlTextureParameterIuiv(name,target,GL_TEXTURE_SWIZZLE_RGBA,swizzle);
+			constexpr GLenum pname[4] = { GL_TEXTURE_SWIZZLE_R, GL_TEXTURE_SWIZZLE_G, GL_TEXTURE_SWIZZLE_B, GL_TEXTURE_SWIZZLE_A };
+			for (uint32_t i = 0u; i < 4u; ++i)
+			{
+				gl->extGlTextureParameteriv(name, target, pname[i], swizzle+i);
+			}
 		}
 
 		void regenerateMipMapLevels() override
 		{
 			if (params.subresourceRange.levelCount <= 1u)
 				return;
 
-			COpenGLExtensionHandler::extGlGenerateTextureMipmap(name,target);
+			assert(false);
+			//COpenGLExtensionHandler::extGlGenerateTextureMipmap(name,target);
```

Comment:
ES doesn't have?

Reply:
this has to be done in a thread, i  think i'll add `regenerateMipMapLevels` to ILogicalDevice

Reply:
actually to ICommandBuffer since in vulkan mipmaps must be generated manually via VkCmdBlitImage
well, cpu2gpu gotta be fun now....
You might wanna talk about this

Reply:
the overarching idea will be one day to implement the Polyphase convolution image filter, and then a compute shader that can take the precomputed polyphase windows for each axis and carry out the mip-mapping by itself.

This will then require us to implement an interface for a GPU-accelerated CPU2GPU converter.

But this is a major TODO for later.